include/wlr/backend.h

```diff
@@ -1,7 +1,29 @@
 #ifndef _WLR_BACKEND_H
 #define _WLR_BACKEND_H
 
-struct wlr_backend *wlr_backend_init();
-void wlr_backend_free(struct wlr_backend *backend);
+#include <wayland-server.h>
+
+struct wlr_backend_impl;
+struct wlr_backend_state;
+
+struct wlr_backend {
+	const struct wlr_backend_impl *impl;
+	struct wlr_backend_state *state;
```

Comment:
No - it's not a fully defined type. I figure each backend can define the type itself in a private header, and then we get better type safety. A bit unorthodox but I think I like this approach.